irohad/ametsuchi/impl/temporary_wsv_impl.cpp

```diff
@@ -78,16 +78,28 @@ namespace iroha {
                            return false;
                          });
           if (not cmd_is_valid) {
-            transaction_->exec("ROLLBACK TO SAVEPOINT savepoint_;");
+            rollbackToSavepoint("savepoint_");
             return expected::makeError(cmd_error);
           }
         }
         // success
-        transaction_->exec("RELEASE SAVEPOINT savepoint_;");
+        releaseSavepoint("savepoint_");
         return {};
       };
     }
 
+    void TemporaryWsvImpl::createSavepoint(const std::string &name) {
```

Comment:
It's barely safe. Return please new object with RAII behavior, such that dtor calls rollback, unless specified explicit release